addons/website_slides/models/slide_slide.py

```diff
@@ -560,11 +560,17 @@ def _on_change_document_binary_content(self):
 
     @api.onchange('slide_category')
     def _on_change_slide_category(self):
-        """ Prevents mis-match when ones uploads an image and then a pdf without saving the form. """
+        """ Prevents mis-match when ones uploads an image and then a pdf without saving the form.
+        Also erases the html_content/url when the category should not have a html_content/url"""
         if self.slide_category != 'infographic' and self.image_binary_content:
             self.image_binary_content = False
         elif self.slide_category != 'document' and self.document_binary_content:
             self.document_binary_content = False
+        elif self.slide_category != 'article' and self.html_content:
```

Comment:
I don't think it should be `elif` here, as there could have been an `html_content` and an `attachment` couldn't it? Nothing cleared `html_content` before your fix, and we won't update databases directly. (Maybe in the master we can clean it if we think it's worth it then).